### PR TITLE
Einthusan Presence

### PR DESCRIPTION
**Pull request by Gurrman375**
Einthusan Presence working.
image
image
image
image

### CODE REVIEW

File: Einthusan/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    "id": "446528923535278082"
+  },
+  "url": "einthusan.tv",
+  "description": {
+    "en": "Einthusan: The leading Premium South Asian streaming content provider of full-length feature films in on-demand superior HD. We're expanding to bring you high-bitrate audio albums, movie clips, and music videos. Explore the social entertainment hub where users discover and consume South Asian content at it's best."
+  },
+  "service": "Einthusan",
+  "logo": "https://i.imgur.com/63yIrXo.jpg",
+  "thumbnail": "https://i.imgur.com/Gmrtp1s.png",
+  "color": "#000000",
```
Comment (Alanexei) [inline Einthusan/dist/metadata.json:13]: Maybe try and find a more suitable color.

File: Einthusan/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  "tags": [
```
Comment (Alanexei) [inline Einthusan/dist/metadata.json:14]: Tags must be all lowercase

Comment (Gurrman375) [reply Einthusan/dist/metadata.json:14]: fixed

**Comment by Bas950** [comment]
On Hold - We are changing some rules regarding presence creation, we will update you soon.

**Comment by Bas950** [comment]
Please code your presence in TypeScript (TS).
For more info see:
https://discordapp.com/channels/493130730549805057/607524579874832446/704792826025869393

**Comment by Gurrman375** [comment]
Now has TypeScript

**Comment by Gurrman375** [comment]
image
image
image

File: Einthusan/presence.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+presence.on("UpdateData", async() => {
+    var presenceData = {
+        largeImageKey: "logo",
+        details: "",
+        startTimestamp: 0,
+        smallImageKey: "",
+        smallImageText: "",
+        state: "",
+        endTimestamp: 0
```
Comment (Bas950) [inline Einthusan/presence.ts:30]: remove all, just only set the logo here

File: Einthusan/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "../tsconfig.json",
+    "compilerOptions": {
+      "outDir": "./dist/"
+    }
+  }
```
Comment (Bas950) [inline Einthusan/tsconfig.json:6]: https://docs.premid.app/en/dev/presence#filling-in-the-tsconfigjson-file

outdir has a / too much

File: Einthusan/presence.ts (hunk 3 above)
Comment (Gurrman375) [reply Einthusan/presence.ts:30]: If I do that I get 74 errors.
image

Comment (Bas950) [reply Einthusan/presence.ts:30]: are you not editting this in your local repo (so you have the index files aswell)

Comment (Gurrman375) [reply Einthusan/presence.ts]: This is my local repo. The errors revolve around details, startTimestamp, smallImageKey, smallImageText, state, and endTimestamp need to be defined.
image

Comment (Gurrman375) [reply Einthusan/presence.ts]: I do edit in my repo, but for some reason if I don't have the above values trypescript goes crazy.
Javascript does however work fine.
image

Comment (Bas950) [reply Einthusan/presence.ts]: you shouldnt get that issue if you have the index.d.ts files 
image

Comment (Gurrman375) [reply Einthusan/presence.ts]: image

Comment (Bas950) [reply Einthusan/presence.ts]: nvm I know why:
var presenceData = {
        largeImageKey: "logo"
    };
should be:
var presenceData: presenceData = {
        largeImageKey: "logo"
    };

Comment (Gurrman375) [reply Einthusan/presence.ts]: Found the fix
image

**Pull request merged**